pkg/ext/delegate_error_test.go

```diff
@@ -0,0 +1,1109 @@
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				listOptions: &metainternalversion.ListOptions{},
+			},
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {
+				ms.EXPECT().List(gomock.Any(), gomock.Any()).Return(&TestTypeList{}, nil)
+			},
+			wantErr: false,
+			expect: output{
+				listResult: &TestTypeList{},
+				err:        nil,
+			},
+		},
+		{
+			name: "missing user in the context",
+			input: input{
+				ctx:         context.Background(),
+				listOptions: &metainternalversion.ListOptions{},
+			},
+			storeSetup: func(mockStore *MockStore[*TestType, *TestTypeList]) {},
+			wantErr:    true,
+			expect: output{
+				listResult: nil,
+				err:        errMissingUserInfo,
+			},
+		},
+		{
+			name: "convertListOptions error",
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				listOptions: &metainternalversion.ListOptions{},
+			},
+			storeSetup:           func(mockStore *MockStore[*TestType, *TestTypeList]) {},
+			simulateConvertError: true,
+			wantErr:              true,
+			expect: output{
+				listResult: nil,
+				err:        assert.AnError,
+			},
+		},
+		{
+			name: "error returned by store",
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				listOptions: &metainternalversion.ListOptions{},
+			},
+			storeSetup: func(mockStore *MockStore[*TestType, *TestTypeList]) {
+				mockStore.EXPECT().List(gomock.Any(), gomock.Any()).Return(nil, assert.AnError)
+			},
+			wantErr: true,
+			expect: output{
+				listResult: nil,
+				err:        assert.AnError,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			addToSchemeTest(scheme)
+
+			if !tt.simulateConvertError {
+				scheme.AddConversionFunc(&metainternalversion.ListOptions{}, &metav1.ListOptions{}, convert_internalversion_ListOptions_to_v1_ListOptions)
+			}
+
+			gvk := schema.GroupVersionKind{Group: "ext.cattle.io", Version: "v1", Kind: "TestType"}
+			gvr := schema.GroupVersionResource{Group: "ext.cattle.io", Version: "v1", Resource: "testtypes"}
+
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := NewMockStore[*TestType, *TestTypeList](ctrl)
+			tt.storeSetup(mockStore)
+
+			testDelegate := &delegateError[*TestType, *TestTypeList]{
+				inner: &delegate[*TestType, *TestTypeList]{
+					scheme: scheme,
+					t:      &TestType{},
+					tList:  &TestTypeList{},
+					gvk:    gvk,
+					gvr:    gvr,
+					store:  mockStore,
+				},
+			}
+
+			result, err := testDelegate.List(tt.input.ctx, tt.input.listOptions)
+			if tt.wantErr {
+				// check if we have an error
+				assert.Error(t, err)
+
+				// check if error is an apierror
+				if _, ok := err.(apierrors.APIStatus); ok {
+					assert.True(t, ok)
+					return
+				}
```

Comment:
Should this be

```suggestion
				_, ok := err.(apierrors.APIStatus)
				assert.True(t, ok)
```

? Right now if the error returned is not an APIStatus there just won't be any assert.